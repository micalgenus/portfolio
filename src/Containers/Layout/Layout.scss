/**
  * Font
  */
@font-face {
  font-family: 'BMJUA';
  src: url('https://cdn.jsdelivr.net/gh/projectnoonnu/noonfonts_one@1.0/BMJUA.woff') format('woff');
  font-weight: normal;
  font-style: normal;
}

/**
  * Global CSS
  */

::selection {
  background: #8d7edc;
  color: #fff;
  text-shadow: none;
}
::-moz-selection {
  background: #8d7edc;
  color: #fff;
  text-shadow: none;
}
::-webkit-selection {
  background: #8d7edc;
  color: #fff;
  text-shadow: none;
}

* {
  font-family: 'BMJUA', 'Source Sans Pro';
  color: #333;
  padding: 0;
  margin: 0;
}

a {
  text-decoration: none;
  &:hover,
  &:active {
    text-decoration: none;
  }
}

/**
  * Configuration variable
  */
$header-height: 3em;

/**
  * Layout CSS
  */
header {
  display: flex;
  align-items: center;
  justify-content: center;
  text-align: center;
  width: 100%;
  height: $header-height;
  box-shadow: 0px 1px rgba(0, 0, 0, 0.1);
}

.container {
  margin-top: $header-height;
}
