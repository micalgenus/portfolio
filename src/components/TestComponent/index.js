import React from 'react';
import PropTypes from 'prop-types';

import Enzyme, { shallow } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';

Enzyme.configure({ adapter: new Adapter() });

export default class TestComponent {
  constructor(props) {
    this.props = {
      name: props.name,
      component: props.namcomponente,
      snapshot: props.snapshot === true ? true : false,
      args: props.args || {},
    };
  }

  run = callback => {
    const component = this.props.component;
    const args = this.props.args;

    describe(this.props.name, () => {
      if (this.props.snapshot === true) {
        it('snapshot test', () => {
          const wrapper = shallow(<component {...args} />);
          expect(wrapper).toMatchSnapshot();
        });
      }

      if (typeof callback === 'function') callback.bind(this)();
    });
  };
}

TestComponent.defaultProps = { snapshot: false, args: {} };

TestComponent.propTypes = {
  name: PropTypes.string,
  component: PropTypes.object,
  snapshot: PropTypes.bool,
  args: PropTypes.object,
};

const snapshotTest = (name, component, argv) => {
  if (!name || !component) return null;
  const snapshotTestComponent = new TestComponent({ name: name, component: component, snapshot: true, args: argv });
  snapshotTestComponent.run(null);
};

export { snapshotTest };
