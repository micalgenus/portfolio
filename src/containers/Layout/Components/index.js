import Header from './Header';
import Router from './Router';

export { Header, Router };
