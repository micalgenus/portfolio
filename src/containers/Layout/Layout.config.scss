/**
  * Configuration variable
  */
$header-height: 60px;
