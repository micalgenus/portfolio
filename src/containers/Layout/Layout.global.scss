/**
  * Global CSS
  */

::selection {
  background: #8d7edc;
  color: #fff;
  text-shadow: none;
}
::-moz-selection {
  background: #8d7edc;
  color: #fff;
  text-shadow: none;
}
::-webkit-selection {
  background: #8d7edc;
  color: #fff;
  text-shadow: none;
}

body {
  display: flex;
}

html,
body,
#root {
  width: 100%;
}

* {
  font-family: 'BMJUA', 'Source Sans Pro';
  color: #333;
  padding: 0;
  margin: 0;
  scroll-behavior: smooth;
}

a {
  text-decoration: none;
  &:hover,
  &:active {
    text-decoration: none;
  }
}

li {
  list-style: none;
}

p {
  font-size: 1.2em;
}
