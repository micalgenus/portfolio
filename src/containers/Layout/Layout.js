import React, { Component } from 'react';
import PropTypes from 'prop-types';

import { Header, Router as Routing } from './Components';

import './Layout.scss';

export default class Layout extends Component {
  render() {
    return (
      <Router>
        <div>
          <Header routers={this.props.routers} />
          <Routing routers={this.props.routers} />
        </div>
      </Router>
    );
  }
}

Layout.defaultProps = {
  routers: [],
};

Layout.propTypes = {
  routers: PropTypes.array,
};
