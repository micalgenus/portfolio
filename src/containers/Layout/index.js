import React, { Component } from 'react';
import PropTypes from 'prop-types';
import { observer, inject } from 'mobx-react';
import { BrowserRouter as Router } from 'react-router-dom';

import './Layout.font.scss';
import './Layout.global.scss';

import { Header, Footer, Router as Routing, ScrollTo } from './Components';

@inject('page')
@inject('scroll')
@observer
class Layout extends Component {
  constructor(props) {
    super(props);
    this.events = [
      { event: 'scroll', method: this.handleScroll },
      { event: 'scroll', method: this.handleResize },
      { event: 'resize', method: this.handleResize },
    ];
  }

  // Add Window event
  componentDidMount = () => {
    for (const e of this.events) {
      console.log(e);
      window.addEventListener(e.event, e.method);
    }
  };

  // Remove Window event
  componentWillUnmount = () => {
    for (const e of this.events) window.removeEventListener(e.event, e.method);
  };

  // Update scroll data with mobx
  handleScroll = () => {
    const { changeScrollY } = this.props.scroll;
    changeScrollY(window.scrollY || document.documentElement.scrollTop);
  };

  // Update window height with redux
  handleResize = () => {
    const { changeWindowHeight } = this.props.scroll;
    changeWindowHeight(document.getElementById('root').scrollHeight - document.documentElement.clientHeight);
  };

  render() {
    const { page, scroll } = this.props;

    return (
      <Router>
        <div>
          <Header
            routers={this.props.routers}
            currentPath={page.currentPath}
            changePath={page.changePath}
            startLoadPage={page.startLoadPage}
            movedScrollY={scroll.movedScrollY}
          />
          <Routing routers={this.props.routers} endLoadPage={page.endLoadPage} />
          <Footer />
          <ScrollTo scrollY={scroll.scrollY} height={scroll.height} />
        </div>
      </Router>
    );
  }
}

Layout.defaultProps = {
  routers: [],
};

Layout.propTypes = {
  routers: PropTypes.array.isRequired,
};

export default Layout;
