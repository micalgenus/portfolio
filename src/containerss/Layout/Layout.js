import React, { Suspense, Component } from 'react';
import PropTypes from 'prop-types';
import { BrowserRouter as Router, Route, Switch } from 'react-router-dom';
import { Header } from './Components';

import './Layout.scss';

export default class Layout extends Component {
  render() {
    return (
      <Router>
        <div>
          <Header routers={this.props.routers} />
          <div className="container">
            <Suspense fallback={<div>Loading...</div>}>
              <Switch>
                {this.props.routers.map((route, idx) =>
                  route.Component ? <Route key={idx} exact={route.exact} path={route.path} component={props => <route.Component {...props} />} /> : null
                )}
              </Switch>
            </Suspense>
          </div>
        </div>
      </Router>
    );
  }
}

Layout.defaultProps = {
  routers: [],
};

Layout.propTypes = {
  routers: PropTypes.array,
};
