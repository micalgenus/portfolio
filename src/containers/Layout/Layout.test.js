import { snapshotTest } from '@/TestComponent';
import Layout from './index';
import Routers from '@/Router';

snapshotTest('/Containers/Layout', Layout, { routers: Routers });
