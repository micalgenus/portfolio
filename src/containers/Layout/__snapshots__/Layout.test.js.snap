// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`/Containers/Layout snapshot test 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <component
    routers={
      Array [
        Object {
          "Component": Object {
            "$$typeof": Symbol(react.lazy),
            "_ctor": [Function],
            "_result": null,
            "_status": -1,
          },
          "exact": true,
          "path": "/",
          "title": "홈",
        },
        Object {
          "Component": Object {
            "$$typeof": Symbol(react.lazy),
            "_ctor": [Function],
            "_result": null,
            "_status": -1,
          },
          "path": "/projects",
          "title": "프로젝트",
        },
      ]
    }
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "checkPropTypes": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "routers": Array [
        Object {
          "Component": Object {
            "$$typeof": Symbol(react.lazy),
            "_ctor": [Function],
            "_result": null,
            "_status": -1,
          },
          "exact": true,
          "path": "/",
          "title": "홈",
        },
        Object {
          "Component": Object {
            "$$typeof": Symbol(react.lazy),
            "_ctor": [Function],
            "_result": null,
            "_status": -1,
          },
          "path": "/projects",
          "title": "프로젝트",
        },
      ],
    },
    "ref": null,
    "rendered": null,
    "type": "component",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "routers": Array [
          Object {
            "Component": Object {
              "$$typeof": Symbol(react.lazy),
              "_ctor": [Function],
              "_result": null,
              "_status": -1,
            },
            "exact": true,
            "path": "/",
            "title": "홈",
          },
          Object {
            "Component": Object {
              "$$typeof": Symbol(react.lazy),
              "_ctor": [Function],
              "_result": null,
              "_status": -1,
            },
            "path": "/projects",
            "title": "프로젝트",
          },
        ],
      },
      "ref": null,
      "rendered": null,
      "type": "component",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "legacyContextMode": "parent",
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getChildContext": Object {
            "calledByRenderer": false,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
